core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournal.java

```diff
@@ -365,13 +378,16 @@ public synchronized void resume() {
     Preconditions.checkState(mTailerThread == null, "tailer is not null");
 
     // Cancel and wait for active catch-up thread.
+    // TODO(jiacheng): if the thread is gone we only crash here than immediately?
     if (mCatchupThread != null && mCatchupThread.isAlive()) {
```

Comment:
We check `mCatchupThread.isAlive()` here, is that a potential problem? If the thread is not healthy then we don't cancel it? In this PR i added an extra crash so maybe we don't need to change here.

Reply:
i feel canceling a unhealthy thread doesn't create side effects so the current code looks safe to me.

Reply:
Ok so if the `mCatchupThread` crashes, i'll let the thread itself and its UncaughtExceptionHandler take care of that. The next `catchup()` will just ignore that and just create a new thread (and the old dead thread ref will be overwritten and recycled). The current code seems fine to me. I just added some more comments to this piece of code because there's too much unexplained logic.